src/plugin/adapter-registry/c-tracker.ts

```diff
@@ -58,16 +58,71 @@ export class CTracker extends AdapterVariableTracker {
         }
         this.logger.debug('Resolved', variable.name, { start: variableAddress, size: variableSize });
         const address = BigInt(variableAddress);
+        const startAddress = toHexStringWithRadixMarker(address);
+        const isPointer = this.isMaybePointer(variable, startAddress);
         const variableRange: VariableRange = {
             name: variable.name,
-            startAddress: toHexStringWithRadixMarker(address),
+            startAddress,
             endAddress: variableSize === undefined ? undefined : toHexStringWithRadixMarker(address + variableSize),
             value: variable.value,
             type: variable.type,
+            isPointer,
         };
         return variableRange;
     }
 
+    override async getLocals(session: vscode.DebugSession): Promise<VariableRange[]> {
+        await this.getAddressSpaces(session);
+        return super.getLocals(session);
+    }
+
+    async getAddressSpaces(session: vscode.DebugSession): Promise<void> {
+        let expression = '';
+        switch (session.type) {
+            case 'gdb':
+                expression = '> info proc mappings';
+                break;
+            case 'arm-debugger':
+                expression = '> info memory';
+                break;
+            case 'embedded-debug': // I haven't found documentation on what commands to use for this.
+            default:
+                return;
+        }
```

Comment:
This class is intended to be generic enough to be used with all C debuggers (i.e. anything that understands the `sizeof` and `&` operators). Functionality that is more specific than that should added to a separate `AdapterCapabilities` implementation rather than directly here. In this case, I would say the price of an error - the user seeing a message that no memory can be loaded for the value of a certain variable they've selected - is low enough that I don't think extra effort is warranted to avoid it - let's just let them try the command if they wish.

Reply:
OK, I've removed the validity check.